tests/align/test_alignment_dp_formula.h

```diff
@@ -390,148 +393,173 @@ SEQAN_DEFINE_TEST(test_dp_formula_trace_global_linear_all_direction)
         TTraceValue traceValue = _computeScore(activeCell, prevDiagonal, prevHorizontal, prevVertical, 'A', 'A',
                                                scoringScheme, RecursionDirectionAll(), TDPProfileSingleTrace());
 
-        SEQAN_ASSERT_EQ(traceValue, +TraceBitMap_<>::VERTICAL | TraceBitMap_<>::MAX_FROM_VERTICAL_MATRIX);
+        SEQAN_ASSERT_EQ(static_cast<int>(traceValue), static_cast<int>(TraceBitMap_<>::VERTICAL | TraceBitMap_<>::MAX_FROM_VERTICAL_MATRIX));
         SEQAN_ASSERT_EQ(activeCell._score, 6);
-        SEQAN_ASSERT_EQ(prevDiagonal._score, -10);
+        SEQAN_ASSERT_EQ(prevDiagonal._score, 10);
         SEQAN_ASSERT_EQ(prevHorizontal._score, 10);
-        SEQAN_ASSERT_EQ(prevVertical._score, 10);
+        SEQAN_ASSERT_EQ(prevVertical._score, 6);
 
+        prevDiagonal._score = -10;
+        prevVertical._score = 10;
         traceValue = _computeScore(activeCell, prevDiagonal, prevHorizontal, prevVertical, 'A', 'A',
                                                scoringScheme, RecursionDirectionAll(), TDPProfileCompleteTrace());
 
-        SEQAN_ASSERT_EQ(traceValue, +TraceBitMap_<>::VERTICAL | +TraceBitMap_<>::HORIZONTAL |
+        SEQAN_ASSERT_EQ(static_cast<int>(traceValue), static_cast<int>(TraceBitMap_<>::VERTICAL | TraceBitMap_<>::HORIZONTAL )|
                         TraceBitMap_<>::MAX_FROM_VERTICAL_MATRIX | TraceBitMap_<>::MAX_FROM_HORIZONTAL_MATRIX);
         SEQAN_ASSERT_EQ(activeCell._score, 6);
-        SEQAN_ASSERT_EQ(prevDiagonal._score, -10);
+        SEQAN_ASSERT_EQ(prevDiagonal._score, 10);
         SEQAN_ASSERT_EQ(prevHorizontal._score, 10);
-        SEQAN_ASSERT_EQ(prevVertical._score, 10);
+        SEQAN_ASSERT_EQ(prevVertical._score, 6);
     }
 
     {
+        prevDiagonal._score = -10;
+        prevVertical._score = 10;
         prevHorizontal._score = 0;
         TTraceValue traceValue = _computeScore(activeCell, prevDiagonal, prevHorizontal, prevVertical, 'A', 'A',
                                                scoringScheme, RecursionDirectionAll(), TDPProfileSingleTrace());
 
-        SEQAN_ASSERT_EQ(traceValue, +TraceBitMap_<>::VERTICAL | TraceBitMap_<>::MAX_FROM_VERTICAL_MATRIX);
+        SEQAN_ASSERT_EQ(static_cast<int>(traceValue), static_cast<int>(TraceBitMap_<>::VERTICAL | TraceBitMap_<>::MAX_FROM_VERTICAL_MATRIX));
         SEQAN_ASSERT_EQ(activeCell._score, 6);
-        SEQAN_ASSERT_EQ(prevDiagonal._score, -10);
```

Comment:
why did these scores actually change?

Reply:
because before it was superfluous to do so. Now, I changed some generic functionality for performance reasons.